app/src/main/java/com/pvp/app/service/HealthConnectServiceImpl.kt

```diff
@@ -48,26 +78,37 @@ class HealthConnectServiceImpl @Inject constructor(
         }
     }
 
-    override suspend fun aggregateDistance(
+    override suspend fun aggregateTotalCalories(
         start: Instant,
         end: Instant
     ): Double {
         return try {
             val response = client.aggregate(
                 AggregateRequest(
-                    metrics = setOf(DistanceRecord.DISTANCE_TOTAL),
+                    metrics = setOf(TotalCaloriesBurnedRecord.ENERGY_TOTAL),
                     timeRangeFilter = TimeRangeFilter.between(
                         start,
                         end
                     )
                 )
             )
 
-            response[DistanceRecord.DISTANCE_TOTAL]?.inMeters ?: 0.0
+            response[TotalCaloriesBurnedRecord.ENERGY_TOTAL]?.inCalories ?: 0.0
         } catch (e: Exception) {
-            Log.e("DISTANCE", "EXCEPTION ${e.message}")
-
             0.0
         }
     }
```

Comment:
Aren't all these methods in the end doing the same, except final property (inCalories, inMeters) is different?
If that's the case, could just make one method and reuse it instead of copy/paste. Most likely of private visibility.